user_if/main.c

```diff
@@ -4,22 +4,77 @@
 #include "../error_handler/error_handler.h"
 #include "../globals/globals.h"
 
+char* static_menu(char** devices_list, int num_of_devices);
+void clear_screen();
+void display_banner();
+
 int main (int argc, char** argv) {
     char *devices_list[DEVICE_LIST_MAX_SIZE];
     int num_of_devices = 0;
     int errNum = 0;
+    char *device_selected;
 
     #if _WIN32
     // Initialize library to use local encoding
     errNum = pcap_init(PCAP_CHAR_ENC_LOCAL, errbuf);
     if (errNum == PCAP_ERROR) {
-        display_error("Could not initialize pcap library!");
+        pcap_error("Could not initialize pcap library!");
     }
     #endif
 
     num_of_devices = scan_all_available_devices(devices_list);
 
-    for(int i = 0; i < num_of_devices; i++) {
-        printf("%s\n", devices_list[i]);
+    device_selected = static_menu(devices_list, num_of_devices);
+}
+
+char* static_menu(char** devices_list, int num_of_devices) {
+    int selection = -1;
+
+    clear_screen();
+
+    display_banner();
+
+    //Loop until user selects a valid option
+    while ((selection < 0) || (selection > num_of_devices)) {
```

Comment:
```suggestion
    while ((selection < 0) || (selection >= num_of_devices)) {
```

Reply:
First one is 0.

Reply:
That's why it should be >= instead of >. This while wants to check "while selection is invalid" and if we had for example 6 devices we would have identifiers from 0 to 5 included, therefore if the user selected the ID 6, it would be invalid and therefore the condition of the while would need to include it.

In fact, you inverted the condition to check for a valid selection and wrote `((selection >= 0) && (selection < num_of_devices))`, which is correct. Doing the negated of `((selection < 0) || (selection > num_of_devices))` would be `((selection >= 0) && (selection <= num_of_devices))`.

Reply:
I feel offended by this